xarray/tests/test_ufuncs.py

```diff
@@ -172,6 +172,41 @@ def test_xarray_ufuncs_deprecation():
     with pytest.warns(PendingDeprecationWarning, match='xarray.ufuncs'):
         xu.cos(xr.DataArray([0, 1]))
 
+    with pytest.warns(None) as record:
+        xu.angle(xr.DataArray([0, 1]))
+    record = [el.message for el in record
+              if el.category == PendingDeprecationWarning]
+    assert len(record) == 0
+
+
+@requires_np113
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize(
+    'name',
+    [name for name in dir(xu)
+        if (not name.startswith('_') and hasattr(np, name)
+            and name not in ['print_function', 'absolute_import', 'division'])]
+)
+def test_numpy_ufuncs(name, request):
+    x = xr.DataArray([1, 1])
+
+    np_func = getattr(np, name)
+    if hasattr(np_func, 'nin') and np_func.nin == 2:
+        args = (x, x)
+    else:
+        args = (x,)
+
+    y = np_func(*args)
+
+    if name in ['angle', 'iscomplex']:
+        # these functions need to be handled with __array_function__ protocol
+        assert isinstance(y, np.ndarray)
+    elif name in ['frexp']:
+        # np.frexp returns a tuple
+        assert not isinstance(y, xr.DataArray)
```

Comment:
The output is a tuple of len 2, it's a bit of a special case, not sure if it's really worth adding specific tests for it.